src/main/scala/io/iohk/ethereum/mallet/service/RpcClient.scala

```diff
@@ -0,0 +1,97 @@
+package io.iohk.ethereum.mallet.service
+
+
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+import com.typesafe.config.ConfigFactory
+import io.circe.generic.auto._
+import io.circe.parser._
+import io.circe.syntax._
+import io.circe.{Decoder, Json}
+import io.iohk.ethereum.domain.Address
+import io.iohk.ethereum.jsonrpc.{JsonRpcError, TransactionReceiptResponse}
+import io.iohk.ethereum.mallet.common.{Constants, Err, RpcClientError, Util}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+import scala.util.{Failure, Success, Try}
+
+object RpcClient {
+  // TODO: CL option to enable akka logging
+  private val akkaConfig = ConfigFactory.load("mallet")
+
+  implicit val actorSystem = ActorSystem("mallet_rpc", akkaConfig)
```

Comment:
why not instantiate it in Mallet app and pass to RpcClient as a parameter?

Reply:
I meant to avoid `ActorSystem` leaking out of `RpcClient` cause it's an internal detail. Obviously this is not the case now, because I forgot to add `shutdown()` method to `RpcClient` - then the actor system could be private to `RpcClient`. What do you think about  such approach?

Reply:
To me, actor system is not really internal to rpc client. Even though for mallet it's only used here, it has an impact on the whole app (for example, keeps it alive until it's shutdown), that's why I'd opt for putting in on app-level and pass (perhaps implicitly) to rpc client. It would also make RpcClient's dependencies more clear as well as make it reusable.

Reply:
also, is there any reason why we're using the `actorSystem.dispatcher` instead of global EC or, even better, implicit ec passed as a constructor param?

Reply:
What's wrong with using `dispatcher`?

Reply:
since you're using blocking operation (await), doesn't that mean that one of the threads that akka uses internally will be blocked? I know, in this case it doesn't matter since RpcClient is the only component using actors, but for the reasons mentioned above (reusability) I think it would be better to let the user of this class choose which EC he prefers (or, alternatively use global ec that won't block system dispatcher's thread).